### PR TITLE
CIV-9453 - LiP upload evidence to application

### PR DESCRIPTION
**Pull request by neeta-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9453

### Change description ###
Added 3 new screens in general application journey.
image

image
image

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import {CivilServiceClient} from 'client/civilServiceClient';
+import { FormValidationError } from 'common/form/validationErrors/formValidationError';
+import { t } from 'i18next';
+
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('claimantResponseService');
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClientForDocRetrieve: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl, true);
+
+export const getSummaryList = async (formattedSummary: SummarySection, redisKey: string, claimId: string): Promise<void> => {
```
Comment (laarmada) [inline src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts:22]: where is the related test? don't we create more service test?

File: src/main/views/features/generalApplication/upload_documents.njk [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{% from "../../macro/heading-with-caption.njk" import headingWithCaption %}
+{% from "../../macro/csrf.njk" import csrfProtection %}
+{% from "../../macro/error-summary.njk" import errorSummary %}
+{% from "../../macro/button.njk" import addButton %}
+
+{% block pageTitle %}
+  {{ setPageTitle(t, 'PAGES.GENERAL_APPLICATION.UPLOAD_DOCUMENTS.TITLE') }}
+{% endblock %}
+
+{% block content %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/upload_documents.njk:18]: should we set under nested content?

File: src/main/views/features/generalApplication/hearing_arrangements.njk [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{% from "../../macro/contact-us-for-help.njk" import contactUsForHelp %}
+{% from "../../macro/page-title.njk" import setPageTitle %}
+{% from "../../macro/csrf.njk" import csrfProtection %}
+{% from "../../macro/heading-with-caption.njk" import headingWithCaption %}
+
+{% block pageTitle %}
+  {{ setPageTitle(t, 'PAGES.GENERAL_APPLICATION.HEARING_APPLICATION.TITLE') }}
+{% endblock %}
+
+{% block content %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/hearing_arrangements.njk:12]: should we set under nested content?

File: src/main/views/features/generalApplication/want-to-upload-documents.njk [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+{% from "../../macro/error-summary.njk" import errorSummary %}
+{% from "../../macro/contact-us-for-help.njk" import contactUsForHelp %}
+{% from "../../macro/heading-with-caption.njk" import headingWithCaption %}
+{% from "../../macro/button.njk" import addButton %}
+
+{% block pageTitle %}
+  {{ setPageTitle(t, 'PAGES.GENERAL_APPLICATION.WANT_TO_UPLOAD_DOCUMENTS.TITLE') }}
+{% endblock %}
+
+{% block content %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/want-to-upload-documents.njk:15]: should we set under nested content?

File: src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    index= index+ 1;
+    return formattedSummary.summaryList.rows.push(summaryRow(uploadDocument.caseDocument.documentName, '', constructResponseUrlWithIdParams(claimId, GA_UPLOAD_DOCUMENTS+'?id='+index), 'Remove document'));
+  });
+  return undefined;
+};
+
+export const saveDocumentsToUploaded = async (claimId: string, uploadDocument: UploadGAFiles): Promise<void> => {
+  try {
+    const claim = await getCaseDataFromStore(claimId, true);
+    claim.generalApplication = Object.assign(new GeneralApplication(), claim.generalApplication);
```
Comment (laarmada) [inline src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts:35]: strange, why do you need to do that if we do the same in  getCaseDataFromStore .... ?

File: src/main/services/features/generalApplication/generalApplicationService.ts [hunk 6]
```diff
@@ -97,6 +97,18 @@ export const saveApplicationCosts = async (claimId: string, applicationCosts: Ye
   }
 };
 
+export const saveIfPartyWantsToUploadDoc = async (claimId: string, wantToSaveDoc: YesNo): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/services/features/generalApplication/generalApplicationService.ts]: done

File: src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts (hunk 1 above)
Comment (neeta-hmcts) [reply src/main/services/features/generalApplication/uploadEvidenceDocumentService.ts:21]: added test for service file too.

File: src/main/views/features/generalApplication/hearing_arrangements_guidance.njk [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{% extends "claim-details-tpl-dashboard.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "../../macro/contact-us-for-help.njk" import contactUsForHelp %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/hearing_arrangements_guidance.njk:3]: you can remove the import

File: src/main/views/features/generalApplication/upload_documents.njk [hunk 8]
```diff
@@ -0,0 +1,71 @@
+{% extends "claim-details-tpl-dashboard.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/summary-list/macro.njk" import govukSummaryList %}
+{% from "govuk/components/file-upload/macro.njk" import govukFileUpload %}
+{% from "../../macro/page-title.njk" import setPageTitle %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/upload_documents.njk:5]: you can remove some imports.

File: src/main/views/features/generalApplication/want-to-upload-documents.njk [hunk 9]
```diff
@@ -0,0 +1,25 @@
+{% extends "claim-details-tpl-dashboard.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "../../macro/page-title.njk" import setPageTitle %}
+{% from "../../macro/csrf.njk" import csrfProtection %}
+{% from "../../macro/yesNoGenericForm.njk" import yesNoRadioButton %}
+{% from "../../macro/error-summary.njk" import errorSummary %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/want-to-upload-documents.njk:6]: you can remove some imports.

**Review by laarmada — DISMISSED** [review]

**Review by laarmada — DISMISSED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

**Pull request merged**